pipeline/transfer_validator.py

```diff
@@ -33,220 +33,270 @@ def transfer_validator(args):
 
     fields = ["TT", "TE", "TB", "EE", "EB", "BE", "BB"]
 
-    tf_dict = read_transfer(f"{meta.coupling_directory}/transfer_function.npz")
-
     nmt_binning = meta.read_nmt_binning()
     nl = nmt_binning.lmax+1
     lb = nmt_binning.get_effective_ells()
 
     plot_dir = meta.plot_dir_from_output_dir(meta.coupling_directory)
+
+    filtering_pairs = meta.get_independent_filtering_pairs()
+
     # First plot the transfer functions
-    plt.figure(figsize=(20, 20))
-    grid = plt.GridSpec(7, 7, hspace=0.3, wspace=0.3)
-
-    for id1, f1 in enumerate(fields):
-        for id2, f2 in enumerate(fields):
-            ax = plt.subplot(grid[id1, id2])
-
-            if f1 == "TT" and f2 != "TT":
-                ax.axis("off")
-                continue
-            if f1 in ["TE", "TB"] and f2 not in ["TE", "TB"]:
-                ax.axis("off")
-                continue
-            if f1 in ["EE", "EB", "BE", "BB"] and \
-               f2 not in ["EE", "EB", "BE", "BB"]:
-                ax.axis("off")
-                continue
-
-            ax.set_title(f"{f1} $\\rightarrow$ {f2}", fontsize=14)
-
-            ax.errorbar(
-                lb, tf_dict[f1, f2], tf_dict[f1, f2, "std"],
-                marker=".", markerfacecolor="white",
-                color="navy")
-
-            if not ([id1, id2] in [[0, 0], [2, 1],
-                                   [2, 2], [6, 3],
-                                   [6, 4], [6, 5],
-                                   [6, 6]]):
-                ax.set_xticks([])
-            else:
-                ax.set_xlabel(r"$\ell$", fontsize=14)
-
-            if f1 == f2:
-                ax.axhline(1., color="k", ls="--")
-            else:
-                ax.axhline(0, color="k", ls="--")
-
-            ax.set_xlim(meta.lmin, meta.lmax)
-
-    plt.savefig(f"{plot_dir}/transfer.pdf", bbox_inches="tight")
+    for ftag1, ftag2 in filtering_pairs:
+        tf_dict = read_transfer(
+            f"{meta.coupling_directory}/transfer_function_{ftag1}x{ftag2}.npz"
+        )
+
+        plt.figure(figsize=(20, 20))
```

Comment:
Should we externalize the TF plotting, e.g. use [this](https://github.com/simonsobs/SOOPERCOOL/blob/887a49a56d38014d89a189c207d6931da3465623/soopercool/utils.py#L427) function?

Reply:
Indeed. This function may be move to a separate utils module in the future.